File: src/main/java/com/kingmang/ixion/ast/WhileStatement.java
package com.kingmang.ixion.ast;

import com.kingmang.ixion.StatementVisitor;
import com.kingmang.ixion.lexer.Position;

public final class WhileStatement extends Statement {
    public final BlockStatement block;
    public final Expression condition;

    public WhileStatement(Position pos, Expression condition, BlockStatement block) {
        super(pos);
        this.condition = condition;
        this.block = block;
    }

    public <R> R accept(StatementVisitor<R> visitor) {
        return visitor.visitWhile(this);
    }
}

File: src/main/java/com/kingmang/ixion/ast/WhileStatement.kt
package com.kingmang.ixion.ast

import com.kingmang.ixion.StatementVisitor
import com.kingmang.ixion.lexer.Position

class WhileStatement(pos: Position?, @JvmField val condition: Expression?, @JvmField val block: BlockStatement?) : Statement(pos) {

    override fun <R> accept(visitor: StatementVisitor<R?>?): R? {
        return visitor?.visitWhile(this)
    }
}